src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Out of bout exception ofr value :"
+                    + value + " with index " + index);
+        }
+        grow();
+        System.arraycopy(objects,index,objects,index + 1,objects.length - index - 1);
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int j = 0; j < list.size(); j++) {
+            add(list.get(j));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkOutOfBoundsException(index);
+        return objects[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkOutOfBoundsException(index);
+        objects[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkOutOfBoundsException(index);
+        T element = objects[index];
+        System.arraycopy(objects,index + 1, objects,index, objects.length - index - 1);
+        size--;
+        return element;
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public T remove(T value) {
+        T element;
+        for (int j = 0; j < size(); j++) {
+            if (objects[j] == value || objects[j] != null && objects[j].equals(value)) {
+                element = objects[j];
+                System.arraycopy(objects,j + 1, objects,j, objects.length - j - 1);
+                size--;
+                return element;
+            }
```

Comment:
```suggestion
            if (objects[j] == value || objects[j] != null && objects[j].equals(value)) {
               return  remove(j);
            }
```

we can reuse our `remove(int index)` method here